### PR TITLE
Selective blocking for Firefox

### PR DESCRIPTION
**Pull request by chrmod**

### CODE REVIEW

File: extension-manifest-v3/src/background/exceptions.js [hunk 1]
```diff
@@ -30,15 +30,23 @@ function negateFilter(filter) {
...
+chrome.storage.local.get(['exceptions']).then((result) => {
+  exceptions = new Map(Object.entries(result.exceptions));
+});
+
 chrome.storage.onChanged.addListener(async (changes) => {
   if (!changes['exceptions']) {
     return;
   }
 
+  exceptions = new Map(Object.entries(changes['exceptions'].newValue || {}));
```
Comment (smalluban) [inline extension-manifest-v3/src/background/exceptions.js:44]: This is overkill, as the exceptions is a plain object with keys, no need to transfer it to the Map instance.

File: extension-manifest-v3/src/background/exceptions.js [hunk 2]
```diff
@@ -30,15 +30,23 @@ function negateFilter(filter) {
...
+
   const networkFilters = [];
   const cosmeticFilters = [];
 
-  for (const exception of changes['exceptions'].newValue || []) {
+  for (const exception of Object.values(changes['exceptions'].newValue || {})) {
```
Comment (smalluban) [inline extension-manifest-v3/src/background/exceptions.js:49]: Use `exceptions` here which you defined above

File: extension-manifest-v3/src/utils/trackerdb.js [hunk 3]
```diff
@@ -103,6 +103,8 @@ export function isCategoryBlockedByDefault(categoryId) {
     case 'adult_advertising':
     case 'email':
     case 'site_analytics':
+    case 'unidentified':
```
Comment (smalluban) [inline extension-manifest-v3/src/utils/trackerdb.js:106]: I am not sure, but I think we removed that category, and everything, which does not fit its set to `misc`

**Review by smalluban — CHANGES_REQUESTED** [review]

File: extension-manifest-v3/src/utils/trackerdb.js (hunk 3 above)
Comment (chrmod) [reply extension-manifest-v3/src/utils/trackerdb.js:106]: the extension code sets the `unidentified` category for unknown trackers

**Pull request merged**